embc-app/ClientApp/src/app/volunteer/components/valid-from-to/valid-from-to.component.html

```diff
@@ -1,36 +1,46 @@
-        <input type="text" id="validFromDate" class="form-control" [value]="validFrom">
-        <span class="invalid-feedback">Please enter a valid date.</span>
+      <div class="col">
+        <div class="input-group mb-0">
+          <label for="validFromDate_{{id}}">Date</label>
+          <input class="form-control" placeholder="yyyy-mm-dd" name="validFromDate_{{id}}" ngbDatepicker #fromDate="ngbDatepicker" [firstDayOfWeek]="7" navigation="arrows" [(ngModel)]="wrdForm.fromDate" (click)="fromDate.close();" (ngModelChange)="updateDisplay()" required aria-required="true">
+          <div class="input-group-append">
+            <button class="btn btn-icon picker-patch" (click)="fromDate.toggle()" type="button" tabindex="-1">
+              <i class="material-icons">date_range</i>
+            </button>
+          </div>
+          <span *ngIf="!validDate" class="invalid-feedback" style="display: block;">Please enter a valid date.</span>
+        </div>
       </div>
-      <div class="col-3 pl-0">
-        <label for="validFromTime">Time</label>
-        <input type="time" id="validFromTime" class="form-control">
-        <span class="invalid-feedback">Please enter a valid time.</span>
+      <div class="col-3 pl-0 pr-0">
+        <div class="form-group mb-0">
+          <label for="validFromTime_{{id}}">Time</label>
+          <ngb-timepicker id="validFromTime_{{id}}" [(ngModel)]="wrdForm.fromTime" [meridian]="true" [spinners]="false" (ngModelChange)="updateDisplay()"></ngb-timepicker>
+          <span *ngIf="!validTime" class="invalid-feedback" style="display: block;">Please enter a valid time.</span>
+        </div>
       </div>
       <div class="col-3 pl-0">
-        <label for="numberDays"># Days</label>
-        <select class="form-control" id="numberDays">
-          <option>--</option>
-          <option *ngFor="let n of days">{{n}}</option>
+        <label for="numberDays_{{id}}"># Days</label>
+        <select class="form-control" id="numberDays_{{id}}" [(ngModel)]="wrdForm.days" (change)="updateDisplay()">
+          <option *ngFor="let n of days" [ngValue]="n">
+            {{n}}
+          </option>
         </select>
       </div>
     </div>
   </div>
-  <div class="col-4">
+  <div class="col-3">
     <span class="h3">Valid To</span>
     <div class="row">
       <div class="col-7">
-        <label for="validToDate">Date</label>
-        <input type="text" id="validToDate" class="form-control" [value]="validTo" readonly>
+        <label for="validToDate_{{id}}">Date</label>
+        <input type="text" id="validToDate_{{id}}" class="form-control" [value]="displayDate" readonly>
       </div>
       <div class="col-5 pl-0">
-        <label for="validToTime">Time</label>
-        <input type="time" id="validToTime" class="form-control">
+        <label for="validToTime_{{id}}">Time</label>
+        <input type="text" id="validToTime_{{id}}" class="form-control" [value]="displayTime" readonly>
```

Comment:
This input is too narrow: with 4-digit times, am or pm look like an and pn.
I suggest making the from-date a little narrower.
FUTURE: Look into responsive sizing for these (possibly create future task or let business find this and report it).